thirdeye-spi/src/main/java/ai/startree/thirdeye/spi/auth/ThirdEyeAuthorizer.java

```diff
@@ -37,6 +38,14 @@ boolean authorize(
   // if the implementation does not support multiple namespace, please return a list with the value null Collections.singletonList(null)
   @NonNull List<String> listNamespaces(final ThirdEyePrincipal principal);
 
+  /**
+   * Generates the datasource connection configuration given principal
+   *
+   * @param principal the principal to generate datasource for
+   * @return DataSourceApi
+   */
+  DataSourceApi generateDatasourceConnection(final ThirdEyePrincipal principal);
```

Comment:
this interface is implemented by our customers in custom plugins we don't control.
so we need a default behavior to avoid a breaking change.
I think `return null` is fine.